ariadne/cardano/test/Spec.hs

```diff
@@ -172,3 +178,56 @@ expectedAriadneConfig = defaultAriadneCfg
     defaultWalletConfig = acWallet defaultAriadneCfg
     defaultUpdateConfig = acUpdate defaultAriadneCfg
     defaultHistoryConfig = acHistory defaultAriadneCfg
+
+cardanoKnitSpec :: Spec
+cardanoKnitSpec = do
+  specAdaToCoin
+  specTyCoin
+
+type Components = '[Core, Cardano]
+
+getMaybeCoin :: (Elem Components Cardano, Elem Components Core) => Scientific -> Maybe Coin
+getMaybeCoin x = tpMatcher tyCoin $ toValue @Components @Core $ ValueNumber x
+
+specTyCoin :: Spec
+specTyCoin = describe "Check tyCoin" $ do
+  prop "Accept correct numbers" propTyCoinAccept
+  prop "Refuse incorrect numbers" propTyCoinRefuse
+
+propTyCoinAccept :: Property
+propTyCoinAccept = monadicIO $ do
+  x <- pick $ arbitrary `suchThat`
```

Comment:
Why do you need `monadicIO` here at all? Can't this be described as pure property? Also, I suppose you can simplify `toBoundedInteger` call with `TypeApplications`, but that's some taste stuff.